src/index.js

```diff
@@ -332,8 +332,13 @@ export default class Undo {
    * @param {Number} prevStateLength is the previous state according to this.position.
    * @param {Array} state is the current state according to this.position.
    */
-  insertSkippedBlocks(prevStateLength, state) {
-    for (let i = prevStateLength; i < state.length; i += 1) {
+  async insertSkippedBlocks(prevState, state, index) {
+    if (JSON.stringify(prevState[index - 1]) !== JSON.stringify(state[index - 1])) {
```

Comment:
1. Update the function's documentation.
2. Q: Why is it necessary to parse object to string?

Reply:
2. This is because when comparing objects in JavaScript with the === operator, it only compares the top-level properties of the objects. There are alternatives to compare the entire structure of the objects, and I think the best option in our case is to use JSON.stringify. This alternative is simpler and does not have any limitations for our specific case.

Reply:
@MarioRodriguezS I got it, thanks!